web/models/session.ex

```diff
@@ -0,0 +1,33 @@
+defmodule Yggdrasil.Session do
+  alias Yggdrasil.User
+
+  def login(%{"username" => nil}, repo) do
+    :error
+  end
+
+  def login(params, repo) do
+    user = repo.get_by(User, username: String.downcase(params["username"]))
+    case authenticate(user, params["password"]) do
+      true -> {:ok, user}
+      _    -> :error
+    end
+  end
+
+  defp authenticate(_user, nil) do
+    false
+  end
+
+  defp authenticate(nil, _password) do
+    false
```

Comment:
Yes I agree, was aware of it was juggle a lot last night will look at that in a moment, that's a quick change. We also probably should configure comeonin, the defaults are ok for now I think though.

We also probably should have password reset page...